roles/gpu_operator_wait_deployment/files/fetch_pod_metrics.sh

```diff
@@ -0,0 +1,41 @@
+#! /bin/bash -e
+
+if [[ "${1:-}" == "" || "${1}" == "-h" || "${1}" == "--help" ]]; then
+    cat <<EOF
+Fetch metrics from a Pod
+
+Usage: $0 <POD_PORT> <LABEL_SELECTOR> <NAMESPACE> <PROM_GREP_KEY>
+Example:
+  $0 9400 app=nvidia-dcgm-exporter gpu-operator-resources '#HELP'
+EOF
+    exit 0
+fi
+
+POD_PORT="$1"
+LABEL_SELECTOR="$2"
+NAMESPACE="$3"
+PROM_GREP_KEY="$4"
+
+LOCAL_PORT=9401
+
+pod_name=$(oc get pods -oname -l$LABEL_SELECTOR -n$NAMESPACE | head -1);
+if [ -z "$pod_name" ]; then
+    echo "Failed to find a pod for $LABEL_SELECTOR in namespace $NAMESPACE";
+    exit 10;
+fi;
+
+retry=5;
+timeout 10 oc port-forward ${pod_name} ${LOCAL_PORT}:${POD_PORT} -n $NAMESPACE &
```

Comment:
If I understand it correctly, the tunnel will get cleaned up after 10 seconds regardless of what happens, right?

Reply:
```
oc port-forward ${pod_name} ${LOCAL_PORT}:${POD_PORT} -n $NAMESPACE &
OC_PORT_FWD_PID=$!
trap "kill -9 $OC_PORT_FWD_PID 2>/dev/null" EXIT
```
I didn't push the last changes, I've updated the code with this ^^^
so that we kill the tunnel on exit

Reply:
Wouldn't the `timeout` kill it anyway?

Reply:
I changed it because for the metrics I need to test this twice in a row (DCGM & operator), and the 2nd one failed because the port was still open! (didn't timeout yet :))

Reply:
> Typically timeout is used to protect from long running commands,

`oc port-forward` is a long method, no? it never terminates by itself

Reply:
Yes, that's why I found it surprising/confusing that `timeout` is used on `oc port-forward`, and why I asked for a comment clarifying what it's about